### ISSUE TITLE
[rllib] minimal build

### ISSUE THREAD

**Issue by rusu24edward** [ray-project/ray#10072]
# What is your question?

With `pip3 install ray[rllib]`, I get to `Running setup.py install for opencv-python`, and I can't seem to move past this point, even waiting for hours. I looked up opencv-python to understand what it is. I don't have applications in computer vision, so I don't need it in my build. Is there a way to install ray[rllib] without this (and possible other) dependencies?

**Comment by simon-mo** [comment]
Opencv python was release yesterday with a breaking change forcing building from source. We just merged a PR that pins the version that does not require building opencv from source. 

You can try install our nightly wheel for this. Alternatively, you can just try `pip install ray` and install all necessary libraries yourself: openai gym, tensorflow or PyTorch.

**Comment by skvark** [comment]
Just tuning in here to say that your issues are most likely caused just very old ``pip`` version. Last 3 ``opencv-python`` releases (release notes: https://github.com/skvark/opencv-python/releases) have used ``manylinux2014`` standard which require `pip` version >= 19.3. If ``pip`` is older (it might be easily 2+ years or even 5+ years old if you use ``pip`` from your Linux distribution repos) then the build will fallback to source build or fails completely with very old ``pip`` versions. Upgrading with ``pip install --upgrade pip`` should solve the issue (note that this might also fail if you try to update the system provided ``pip``).

**Comment by rusu24edward** [comment]
Thanks for these responses!

@simon-mo I did try to install myself without dependencies and then went through and installed the individually. I actually modified my local install of the rllib library to remove all the atari stuff that needed opencv. This worked reasonably well until I ran into a problem of not being able to find tensorflow. I believe this is a weirdness in the server I'm working on.

I managed to make it work by [installing through conda](https://docs.ray.io/en/master/installation.html#installing-ray-with-anaconda).
